### ISSUE TITLE
[Vagrant] debops not installed if /vagrant is not mounted

### ISSUE THREAD

**Issue by nqb** [debops/debops#1061]
If you want to use DebOps `Vagrantfile` on a host system [like Windows](https://github.com/debops/debops/blob/40a22fc44c180d084af03f1622e21e871728697c/Vagrantfile#L816-L819) where `/vagrant` is not mounted, `vagrant` will fail and `debops` will not be install.

### Steps to reproduce
1. Disable mount of `/vagrant`
2. Run `vagrant up`

### Actual results
`vagrant` fail

### Expected results
`vagrant` should not fail and `debops` should be install.

### Proposal
`/vagrant` is only necessary if you want to build `debops` from sources. 

When `/vagrant` is not mounted, we could install `debops` from PyPI and use [code already present](https://github.com/debops/debops/blob/40a22fc44c180d084af03f1622e21e871728697c/Vagrantfile#L587-L593) to install master branch using `debops-update`.

**Comment by drybjed** [comment]
Sounds good - we cannot really sync the workdir anyway, since symlinks will be broken. On Windows, Vagrant boxes can be treated as demos or to manage remote hosts; development will probably be done on Linux/macOS platforms anyway.

**Comment by nqb** [comment]
I agree, I will provide a fix quickly.

Would it be possible to backport this change to stable releases ?

**Comment by drybjed** [comment]
Of course, I don't see why not, as long as the changes are compatible.

**Comment by nqb** [comment]
> We could install `debops` from PyPI and use [code already present](https://github.com/debops/debops/blob/40a22fc44c180d084af03f1622e21e871728697c/Vagrantfile#L587-L593) to install master branch using `debops-update`.

To be exact, we should use `debops-update` to checkout **current git branch**. I mean, if I'm on stable 1.1, we should use `debops-update` to install stable-1.1 branch.

**Comment by drybjed** [comment]
You mean if you run `vagrant up` in DebOps monorepo checked out to `stable-1.1` branch, it should run `debops-update` and get the `stable-1.1` branch as well? Obvious way to do this is to pass the current branch/tag the monorepo is checked out from to inside of the Vagrant environment via env variables. I suppose that the `Vagrantfile` could run some `git` commands to get that information if possible and fall back gracefully to `master` oterwise.

I don't remember if `debops-update` supports cloning specific tag/branch directly at the moment, if not, implementing this would probably solve the need to check out the specific branch afterwards.

**Comment by nqb** [comment]
> You mean if you run `vagrant up` in DebOps monorepo checked out to `stable-1.1` branch, it should run `debops-update` and get the `stable-1.1` branch as well?

Exactly !

> I don't remember if `debops-update` supports cloning specific tag/branch directly at the moment, if not, implementing this would probably solve the need to check out the specific branch afterwards.

Not yet but should be easy to implement. This is exactly how I want to handle this issue.

**Comment by nqb** [comment]
> Obvious way to do this is to pass the current branch/tag the monorepo is checked out from to inside of the Vagrant environment via env variables

I took a quick look but we will certainly need to use a specific ruby module like `git` and I'm unsure this module is part of a default Ruby installation.

**Comment by nqb** [comment]
```ruby
$ irb
irb(main):001:0> require 'git'
=> true
irb(main):002:0> g = Git.open(ENV['PWD'])
=> #<Git::Base:0x000055b37bbae198 @logger=nil, @working_directory=#<Git::WorkingDirectory:0x000055b37bbae058 @path="/home/nqb/git/packetfence">, @repository=#<Git::Repository:0x000055b37bbadfe0 @path="/home/nqb/git/packetfence/.git">, @index=#<Git::Index:0x000055b37bbadf40 @path="/home/nqb/git/packetfence/.git/index">>
irb(main):003:0> g.current_branch
=> "feature/build-maintenance-binaries"
```
It works but I need to install previously `ruby-git` package on my Debian Buster.

**Comment by nqb** [comment]
We can still try to run a command through a shell like `git rev-parse --abbrev-ref HEAD` but that will be not "portable" between several OS.

What's your point of view @drybjed  ?

**Comment by drybjed** [comment]
Sounds good to me. DebOps is not really portable either, so I think that if this feature works only on supported platforms, that will be fine.
